src/orange/Data.hh

```diff
@@ -3,15 +3,13 @@
 // See the top-level COPYRIGHT file for details.
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
-//! \file Data.hh
+//! \file orange/Data.hh
 //---------------------------------------------------------------------------//
 #pragma once
 
-#include "base/Collection.hh"
-#include "base/CollectionBuilder.hh"
-#include "base/OpaqueId.hh"
-#include "geometry/Types.hh"
-#include "sim/Types.hh"
+#include "corecel/OpaqueId.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/data/CollectionBuilder.hh"
 
 #include "Types.hh"
```

Comment:
Shouldn't those #include also include the subdirectory?
```suggestion
#include "orange/Types.hh"
```

Reply:
I think it's fine to allow relative includes in most places. The auto-update script prefers relative includes in:
- the same directory
- the parent directory (if referencing from a `detail/` directory)

so for example, we don't write `../celeritas/Types.hh` but `universe/Types.hh` is OK.

Reply:
Indeed this works perfectly same the preprocessor looks first in the same directory.  The only __minor__ downside is that searching (grepping) for a specific header is a tad bit more complicated: instead of `grep -r '#include "orange/Types" src` it is `grep -r '#include "orange/Types" src; grep -r '#include "Types.hh"' src/orange` and (for `Types.hh` which has many duplicates) the slight cognitive overhead to answer `which Types.hh` is that).

Either way, I think it should stay as is in this PR and could be updated in another (no functional change) PR.

Reply:
Yeah, this is perhaps also related to the problem of "too many `Types` and `Data` headers"—I've at least consolidated a few of the Types classes :) which are just an artifact of how the codebase has grown.